lib/ruby_smb/generic_packet.rb

```diff
@@ -35,15 +35,20 @@ def packet_smb_version
     end
 
     def status_code
+      value = -1
       smb_version = packet_smb_version
       case smb_version
         when 'SMB1'
-          status_code = WindowsError::NTStatus.find_by_retval(self.smb_header.nt_status.value).first
+          value = self.smb_header.nt_status.value
         when 'SMB2'
-          status_code = WindowsError::NTStatus.find_by_retval(self.smb2_header.nt_status.value).first
-        else
-          nil
+          value = self.smb2_header.nt_status.value
+      end
+      status_code = WindowsError::NTStatus.find_by_retval(value).first
+      if status_code.nil?
+        status_code = WindowsError::NTStatus.new("0x#{value.to_s(16)}", value,
```

Comment:
This should have been WindowsError::ErrorCode.new instead, could you pease try that?